MathLib/LinAlg/Eigen/EigenLinearSolver.cpp

```diff
@@ -119,8 +120,35 @@ class EigenIterativeLinearSolver final : public EigenLinearSolverBase
 
 private:
     T_SOLVER _solver;
+    void setRestart(int const /*restart*/) {
+    }
 };
 
+template <>
+void EigenIterativeLinearSolver<
+    Eigen::GMRES<EigenMatrix::RawMatrixType,
+                 Eigen::IdentityPreconditioner>>::setRestart(int const restart)
+{
+    _solver.set_restart(restart);
+    INFO("-> set restart value: {:d}", _solver.get_restart());
+}
+
+template <>
+void EigenIterativeLinearSolver<Eigen::GMRES<
+    EigenMatrix::RawMatrixType,
+    Eigen::DiagonalPreconditioner<double>>>::setRestart(int const restart)
+{
+    _solver.set_restart(restart);
+    INFO("-> set restart value: {:d}", _solver.get_restart());
+}
+
+template <>
+void EigenIterativeLinearSolver<Eigen::GMRES<EigenMatrix::RawMatrixType, Eigen::IncompleteLUT<double>>>::setRestart(int const restart)
+{
+    _solver.set_restart(restart);
+    INFO("-> set restart value: {:d}", _solver.get_restart());
+}
```

Comment:
Why is the setting of the `restart` value only possible for `IdentityPreconditioner`, `DiagonalPreconditioner`, and `IncompleteLUT`? The restart should be independent from the preconditioner at all.

Reply:
Not sure if the partial specialization will work here.
@joergbuchwald Please try it.

Reply:
That's true. The problem I had, was with the T_SOLVER type. It is not instantiated via a general preconditioner type but via the specific preconditioners. Therefore, I had to make the code specialization for each preconditioner.

Reply:
Something like:
```c++
template <typename Preconditioner>
￼void EigenIterativeLinearSolver<Eigen::GMRES<EigenMatrix::RawMatrixType, Preconditioner>>::setRestart(int const restart)
```